manifests/init.pp

```diff
@@ -29,8 +30,10 @@
   }
 
   file { $confdir:
-    ensure => directory,
-    mode   => '0755',
+    ensure  => directory,
+    mode    => '0755',
+    recurse => $purge_confdir,
```

Comment:
Yeah, `purge => true` only works with `recurse => true`. Perhaps you were thinking of a file resource with a `source` of a remote directory and `recurse => true` being slow?